frontend/src/routes/Governance/components/ViewDiffApiCall.tsx

```diff
@@ -0,0 +1,102 @@
+/* Copyright Contributors to the Open Cluster Management project */
+
+import { useEffect, useState } from 'react'
+import { fireManagedClusterView } from '../../../resources'
+import { resultsTableData } from '../policies/policy-details/PolicyDetailsResults'
+import { useTranslation } from '../../../lib/acm-i18next'
+import { Alert, Button, Divider, Modal, ModalVariant, Skeleton, Title } from '@patternfly/react-core'
+import Highlight from 'react-highlight'
+import './github.min.css'
+import { Grid } from '@mui/material'
+
+export function ViewDiffApiCall({ item }: Readonly<{ item: resultsTableData }>) {
+  const { templateName, apiVersion, kind, cluster } = item
+  const [templateErr, setTemplateError] = useState<string>()
+  const [isFetching, setIsFetching] = useState<boolean>(false)
+  const [relatedObjs, setRelatedObjs] = useState<any[]>()
+  const [isModalOpen, setIsModalOpen] = useState(false)
+
+  const { t } = useTranslation()
+  const handleModalToggle = () => {
+    setIsModalOpen((prevIsModalOpen) => !prevIsModalOpen)
+  }
+  useEffect(() => {
+    if (kind === 'IamPolicy' && apiVersion.includes('policy.open-cluster-management.io')) {
+      setTemplateError(t('IamPolicy is no longer supported'))
+      // finish useEffect
+      return
+    }
+
+    if (isModalOpen) {
+      setIsFetching(true)
+      setRelatedObjs(undefined)
+      setTemplateError(undefined)
+
+      fireManagedClusterView(cluster, kind, apiVersion, templateName, cluster)
+        .then((viewResponse) => {
+          if (viewResponse?.message) {
+            setTemplateError(viewResponse.message)
+          } else {
+            setRelatedObjs(viewResponse.result.status?.relatedObjects)
+          }
+          setIsFetching(false)
+        })
+        .catch((err) => {
+          console.error('Error getting resource: ', err)
+          setTemplateError(err)
+          setIsFetching(false)
+        })
+    }
+  }, [isModalOpen, kind, apiVersion, templateName, cluster, t])
+
+  return (
+    <>
+      <span> {'-'} </span>
+      <div style={{ display: 'inline-block' }}>
+        <Button variant="link" isInline onClick={handleModalToggle} width={50}>
+          {t('View Diff')}
+        </Button>
+      </div>
+
+      <Modal
+        bodyAriaLabel="policy difference modal"
+        tabIndex={0}
+        header={
+          <>
+            <Title headingLevel="h1">Difference for {item.templateName}</Title>
+            <Divider />
+          </>
```

Comment:
I like reusing the template title here
```suggestion
          <>
            <Title headingLevel="h1">
              <TemplateDetailTitle policyKind={item.kind} templateName={item.templateName} compliant="NonCompliant" />
            </Title>
            <Divider style={{ marginTop: '10px' }} />
          </>
```